fw/http_stream_sched.h

```diff
@@ -27,18 +27,22 @@
 #include "http_types.h"
 
 /**
- * @total_weight - total weight of the streams for this scheduler;
- * @active_cnt	 - count of active child streams for this scheduler;
- * @parent	 - parent scheduler;
- * @active	 - root of the active streams scheduler ebtree;
- * @blocked	 - root of the blocked streams scheduler ebtree;
+ * @total_weight  - total weight of the streams for this scheduler;
+ * @active_cnt	  - count of active child streams for this scheduler;
+ * @owner	  - stream that owns this scheduler
+ * @parent	  - parent scheduler;
+ * @active	  - root of the active streams scheduler ebtree;
+ * @blocked	  - root of the blocked streams scheduler ebtree;
+ * @in_empty_list - entry in list of empty schedulers;
  */ 
 typedef struct tfw_stream_sched_entry_t {
 	u64				total_weight;
 	long int			active_cnt;
-	struct tfw_stream_sched_entry_t	*parent;
+	TfwStream 			*owner;
+	struct tfw_stream_sched_entry_t *parent;
 	struct eb_root			active;
 	struct eb_root			blocked;
+	struct list_head		in_empty_list;
```

Comment:
Could we get rid off `in_empty_list` if we will store the pointer to the next entry in the current entry? I mean that while `TfwStreamSchedEntry` in the freelist it casted to `void *` that points to the next `TfwStreamSchedEntry` when we remove it from the list, we zeroes the pointer and cast to `TfwStreamSched` again. In other words when `TfwStreamSchedEntry` in empty list it's casted to linked list.